enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -99,259 +116,217 @@ This is often required for industries such as the US public sector, healthcare o
+* Optimize performance bottlenecks.
 
 ### Graduation Criteria
 
-##### Tech Preview
-
-* Log forwarding is only enabled when annotated with: `clusterlogging.openshift.io/logforwardingtechpreview: enabled`
-* Ability to store logs to the cluster managed log aggregator without defining log forwarding (e.g. status quo)
-* Explicitly define and document log sources and their meanings (e.g. `logs.apps`, `logs.infra`)
-* Explicitly define supported outputs (e.g. `elasticsearch`, `fluentd`)
-- End user documentation, relative API stability
-- Sufficient test coverage
-- Gather feedback from users rather than just developers
-
-##### Tech Preview -> GA 
+##### Tech Preview -> GA
 
-- More testing (upgrade, downgrade, scale)
-- Sufficient time for feedback
+Essential:
+- Rework TP implementation to implement new API - structurally close, much renaming.
+- Implement new GA output types.
+- Sufficient test coverage (upgrade, downgrade, scale)
 - Available by default without tech preview annotation
-- Concrete API
-- Additional supported outputs (e.g. `Splunk`, `Graylog`, `kafka`)
-- Support for SRE usecase: As an administrator of a dedicated cluster, I need to allow dedicated-admins to configure the destination of container logs (logs.apps) separately from infra logs so they are prevented from modifying where infra logs (logs.infra) are forwarded
-- Modify API to support: include and exclude of logs
+- End user documentation.
+
+Nice to have:
+- TP migration strategy (see Upgrade section below)
+- Configurable reconnect (TP already has a fixed reconnect strategy)
 
 ### Upgrade / Downgrade Strategy
 
 #### Upgrade
 Upgrades should be seamless to users of cluster logging deployments managed by the `cluster-logging-operator`.  Admins who make no changes will see no difference after upgrade when this feature is deployed.
 
-Upon upgrade, the `cluster-logging-operator` will continue to route `logs.infra` and `logs.app` logs to the internal logstore if:
-
-* `spec.logstore` is defined
-
-
-The `cluster-logging-operator` will honor the `LogForwarding` resource named `instance` if it exists and the `ClusterLogging` resource named `instance` is properly annotated.
+For users who deployed the TP, we need to decide our migration strategy:
 
+1. Do nothing, require a hard cut-over. GA version will not deploy if TP is present.
+2. Allow TP and GA APIs to co-exist for a time (more work)
+   - controller creates the illusion of deploying both
+   - but actually we munge both configs into a single new-style CR.
 
 #### Downgrade
 Downgrades should be discouraged unless we know for certain the Elasticsearch version managed by cluster logging is the same version.  There is risk that Elasticsearch may have migrated data that is unreadable by an older version.
 
-
 ### Version Skew Strategy
 
 Version skew is not relevant to cluster logging because it is deployed as an OLM managed operator.  Component versions are set in a versioned operator deployment.
```

Comment:
no part of this PR, but version skew does matter because you have to handle upgrades during which:

1) the operator+operand versions will be temporarily mismatched
2) the elastic operator + logging operator (and their respective operands) will be temporarily mismatched

Reply:
Added that note to the doc.